### DISCUSSION TITLE
Sponsor button embed ?

### DISCUSSION THREAD

**Discussion by gpotter2** [community/community#21640] [New to GitHub]
<p>Hi,<br>
Is it possible to embed the “Sponsor” button in a external webpage ? (That would be great)</p>
<p>Also, it would be awesome if it (in the same way it does on GitHub, when multiple sponsors are specified) allowed to specify multiple people. Clicking on it would show the same popup than in GitHub or redirect to the corresponding GitHub (not currently possible because there are no direct link to the sponsor button of a repository).</p>
<p>Thanks for your time</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Thanks for the feedback! I’ve taken your suggestion and passed it along to the appropriate teams. Thanks again for reaching out :grinning:</p>

**Comment by gpotter2** [comment]
<p>To people looking for this: you can hack your way through by using the undocumented “funding_links” page. For instance</p>
<p><a href="https://github.com/secdev/scapy/funding_links?fragment=1" rel="nofollow noopener">https://github.com/secdev/scapy/funding_links?fragment=1</a></p>
<p>But then you’ll need to import github css. You can probably download it from the source if it’s not copyrighted (I don’t expect things like <a href="https://github.githubassets.com/assets/frameworks-146fab5ea30e8afac08dd11013bb4ee0.css" rel="nofollow noopener">https://github.githubassets.com/assets/frameworks-146fab5ea30e8afac08dd11013bb4ee0.css</a> to remain live.) Maybe I’ll figure out a static link / gist someday.</p>

**Comment by nholden** [comment]
<p>:wave: Hi, <a class="mention" href="/gpotter2">@gpotter2</a>! Wanted to let you know that you can now embed a sponsor button on an external web page. To try it out, log into your Sponsors dashboard, and under “Share it out,” click “Embed it.”</p>

**Comment by gpotter2** [comment]
<p>That’s very cool thanks !<br>
Do you know if this also applies to repository-specific sponsor buttons ?</p>

**Comment by nholden** [comment]
<p>We haven’t built that yet, but it’s a great idea!</p>
<p>One approach you could try to build a repository-specific sponsor button that could be a little more resilient and flexible than using that <code>funding_links</code> HTML fragment you mentioned above is to fetch the data from the GraphQL API. Each <code>Repository</code> has a <code>fundingLinks</code> field that you can use like this.</p>
<pre><code class="lang-auto">query { 
  repository(name: "scapy", owner: "secdev") { 
    fundingLinks {
      platform
      url
    }
  }
}
</code></pre>
<p>That would return data that looks like this.</p>
<pre><code class="lang-auto">{
  "data": {
    "repository": {
      "fundingLinks": [
        {
          "platform": "GITHUB",
          "url": "https://github.com/gpotter2"
        },
        {
          "platform": "GITHUB",
          "url": "https://github.com/guedou"
        },
        {
          "platform": "GITHUB",
          "url": "https://github.com/p-l-"
        }
      ]
    }
  }
}
</code></pre>
<p>You can give it a shot yourself in the <a href="https://developer.github.com/v4/explorer/" rel="nofollow noopener">GraphQL API explorer</a> if you’re interested.</p>
<p>If you really dig the CSS for the button, it’s open source as part of <a href="https://github.com/primer/css" rel="nofollow noopener">Primer</a>, which you’re free to pull into your project.</p>
<p>Thanks again for the idea! :sparkles:</p>

**Comment by Slion** [comment]
<p><a class="mention" href="/nholden">@nholden</a> Any chance of getting a dark version, or better even one with configurable color?<br>
Works great otherwise thanks.</p>

**Comment by nikolaydubina** [comment]
yeah, current embeddings and iframes are unusable
- size is too big.
- size is not adjustable.
- size does not match standard "small star" button
- color is black, and cannot change

<img width="535" alt="image" src="https://github.com/community/community/assets/2933061/d99181db-5284-4c2e-b4bc-34573b07f16c">

...embedding large view, is even worse.

**Comment by kmturley** [comment]
For anyone looking for the official solution (which has limitations):

1) Go to your sponsors dashboard
https://github.com/sponsors/username/dashboard

2) Click on Share it out -> Embed it
<img width="693" alt="Screenshot 2024-09-19 at 9 12 05 PM" src="https://github.com/user-attachments/assets/7e56c65f-a7b7-4397-821e-f18df96a0fb9">

3( Copy/paste the embed code you want to use
<img width="643" alt="Screenshot 2024-09-19 at 9 12 15 PM" src="https://github.com/user-attachments/assets/66fb963a-dafe-4e7c-bb6d-c89a4fbeb62d">
